holopy/inference/result.py

```diff
@@ -135,6 +120,24 @@ def add_attr(self, kwargs):
             setattr(self, key, val)
             self._kwargs_keys.append(key)
 
+    def forward(self, pars):
+        if hasattr(self.data, 'original_dims'):
```

Comment:
Oof! This is a reason why, down the line, we should address #230 and ensure that `detector_grid`, `detector_points`, and `subset_data` all play the same way. Not really a comment on this PR though :smiley:

Reply:
This isn't as bad as you think! It's checking if the result was calculated only on a subset of pixels, but to display the hologram you want to calculate it all the pixels in the original image, even if the results object doesn't know about them.

I don't think we can remove the if-check here without assigning some properties to the objects storing holograms, which would mean not using raw xarray.DataArrays. I agree that's a valid long-term goal but this isn't as straightforward as streamlining the functions to generate them.